src/main/java/astvisitors/SymbolTableVisitor.java

```diff
@@ -12,27 +12,33 @@
 import symboltable.SymbolTableHandler;
 
 public class SymbolTableVisitor extends AstVisitor {
-    private SymbolTableHandler symbolTableHandler;
+    private SymbolTableHandler symbolTableHandlerVariables;
+    private SymbolTableHandler symbolTableHandlerFunctions;
 
     public SymbolTableVisitor(boolean printDcl) {
-        this.symbolTableHandler = new SymbolTableHandler(printDcl);
+        this.symbolTableHandlerVariables = new SymbolTableHandler(printDcl);
+        this.symbolTableHandlerFunctions = new SymbolTableHandler(printDcl);
     }
 
     public SymbolTableVisitor() {
-        this.symbolTableHandler = new SymbolTableHandler(false);
+        this.symbolTableHandlerVariables = new SymbolTableHandler(false);
+        this.symbolTableHandlerFunctions = new SymbolTableHandler(false);
     }
 
     @Override
     public void visit(StartNode node) {
-        symbolTableHandler.openScope();
+        symbolTableHandlerVariables.openScope();
+        symbolTableHandlerFunctions.openScope();
         node.getDcls().accept(this);
         node.getStmts().accept(this);
-        symbolTableHandler.closeScope();
+        symbolTableHandlerVariables.closeScope();
+        symbolTableHandlerFunctions.closeScope();
```

Comment:
Could you make a `private void closeScope` function which closes the scope of both symboltables?